lang/intrinsic.md

```diff
@@ -0,0 +1,73 @@
+# Intrinsics
+
+```rust
+pub enum Intrinsic {
+    Exit,
+    PrintStdout,
+    PrintStderr,
+}
+
+impl<M: Memory> Machine<M> {
+    #[specr::argmatch(intrinsic)]
+    fn eval_intrinsic(
+        &mut self,
+        intrinsic: Intrinsic,
+        arguments: List<Value<M>>,
+        ret: Option<Place<M>>,
+    ) -> NdResult;
+}
+```
+
+We start with the `Exit` intrinsic.
+
+```rust
+impl<M: Memory> Machine<M> {
+    fn eval_intrinsic(
+        &mut self,
+        Intrinsic::Exit: Intrinsic,
+        arguments: List<Value<M>>,
+        ret: Option<Place<M>>,
+    ) -> NdResult {
+        throw_machine_stop!("0")
+    }
+}
+```
+
+And there are the `PrintStdout` and `PrintStderr` intrinsics.
+
+```rust
+impl<M: Memory> Machine<M> {
+    fn eval_intrinsic(
+        &mut self,
+        Intrinsic::PrintStdout: Intrinsic,
+        arguments: List<Value<M>>,
+        ret: Option<Place<M>>,
+    ) -> NdResult {
+        self.eval_print(&mut std::io::stdout(), arguments, ret)?
+    }
+
+    fn eval_intrinsic(
+        &mut self,
+        Intrinsic::PrintStderr: Intrinsic,
+        arguments: List<Value<M>>,
+        ret: Option<Place<M>>,
+    ) -> NdResult {
+        self.eval_print(&mut std::io::stderr(), arguments, ret)?
+    }
+
+    fn eval_print(
+        &mut self,
+        stream: &mut impl std::io::Write,
```

Comment:
Oh the Coq translator will not like this type... but I guess we can figure this out when we get there. (Probably we'll have to add print/eprint functions to specr so that different specr backends can have their own implementation.)